### PR TITLE
Avoid generating fields/methods for opaque objects

### PR DESCRIPTION
**Pull request by jhod0**
WIP fix for issue #929 

With libclang 4, passes all tests on my machine except `dump_preprocessed_input`, but that fails on the master branch as well

### CODE REVIEW

**Comment by fitzgen** [comment]
> With libclang 4, passes all tests on my machine except dump_preprocessed_input, but that fails on the master branch as well

Hm... It doesn't fail for me with libclang 4, nor on CI. Can you share what your local failures look like?

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -1654,11 +1658,12 @@ impl CodeGenerator for CompInfo {
...
-        // Yeah, sorry about that.
-        let is_opaque = item.is_opaque(ctx, &());
         if is_opaque {
-            fields.clear();
-            methods.clear();
+            // Opaque item should not have generated methods, fields
+            debug_assert!(fields.is_empty());
+            debug_assert!(methods.is_empty());
+            // fields.clear();
+            // methods.clear();
```
Comment (fitzgen) [inline src/codegen/mod.rs]: Nitpick: these commented out lines should be removed.

File: src/codegen/mod.rs [hunk 2]
```diff
@@ -1654,11 +1658,12 @@ impl CodeGenerator for CompInfo {
             fields.push(field);
         }
 
-        // Yeah, sorry about that.
-        let is_opaque = item.is_opaque(ctx, &());
         if is_opaque {
-            fields.clear();
-            methods.clear();
+            // Opaque item should not have generated methods, fields
```
Comment (fitzgen) [inline src/codegen/mod.rs]: Nitpick: missing a period at the end of this sentence.

File: src/codegen/mod.rs (hunk 1 above)
Comment (jhod0) [reply src/codegen/mod.rs]: Done!

Comment (jhod0) [reply src/codegen/mod.rs]: Done!

**Comment by fitzgen** [comment]
@bors-servo r+

Thanks again @jhod0 !

**Comment by jhod0** [comment]
As for the failing test, I had forgotten to add clang to my PATH. I added it and the test passes

**Comment by fitzgen** [comment]
> As for the failing test, I had forgotten to add clang to my PATH. I added it and the test passes

Ah, great to hear that :)

**Pull request merged**